src/Serialization/Address.purs

```diff
@@ -1,376 +1,283 @@
--- | >>>       { payment: ed25519KeyHashCredType
--- | >>>       , delegation: scriptHashCredType
--- | >>>      "addr1..."
--- | '''
--- |
--- | If the string doesn't encode the address matching given credential types
--- | the function will return `Nothing`
--- |
-baseAddressFromBech32 :: forall p d. { payment :: CredType p, delegation :: CredType d } -> Bech32String -> Maybe (BaseAddress p d)
-baseAddressFromBech32 checks bchString = do
-  addrCsl <- _baseAddressFromBech32Impl maybeFfiHelper bchString
-  _headerCheckBaseAddr maybeFfiHelper checks intToNetTag addrCsl
-
--- | Build address out of its bech32 representation.
--- | Works analogically to `baseAddressFromBech32`.
--- | For example:
--- |
--- | ```purescript
--- | >>> let addr =
--- | >>>      rewardAddressFromBech32
--- | >>>       { payment: ed25519KeyHashCredType }
--- | >>>      "addr1..."
--- | '''
--- |
-rewardAddressFromBech32 :: forall p. { payment :: CredType p } -> Bech32String -> Maybe (RewardAddress p)
-rewardAddressFromBech32 checks bchString = do
-  addrCsl <- _rewardAddressFromBech32Impl maybeFfiHelper bchString
-  _headerCheckRewardAddr maybeFfiHelper checks intToNetTag addrCsl
-
--- | Build `PubKeyAddress` from a single credential.
-pubKeyAddress :: NetworkTag -> Ed25519KeyHash -> PubKeyAddress
-pubKeyAddress nt pkh = BaseAddress { network: nt, payment: pkh, delegation: pkh }
-
--- | Build `ScriptAddress` from a single credential.
-scriptAddress :: NetworkTag -> ScriptHash -> ScriptAddress
-scriptAddress nt sh = BaseAddress { network: nt, payment: sh, delegation: sh }
-
----- cred types
-
--- | Sets BaseAddress type arguments during decoding.
-ed25519KeyHashCredType :: CredType Ed25519KeyHash
-ed25519KeyHashCredType = CredType "from_keyhash"
-
--- | Sets BaseAddress type arguments during decoding.
-scriptHashCredType :: CredType ScriptHash
-scriptHashCredType = CredType "from_scripthash"
-
--- | NetworkTag accessor on an Address
-networkTag
-  :: forall a p d
-   . Address a p d
-  => a
-  -> NetworkTag
-networkTag = addrR >>> _.network
-
--- | NetworkTag accessor on an Address
-networkTagInt
-  :: forall a p d
-   . Address a p d
-  => a
-  -> Int
-networkTagInt = addrR >>> _.network >>> netTagToInt
-
--- | Payment credential accessor on an Address
-paymentCred
-  :: forall a p x
-   . Address a p x
-  => a
-  -> p
-paymentCred = addrR >>> _.payment
-
--- | Delegation credential accessor on an Address
-delegationCred
-  :: forall a x d
-   . Address a x d
-  => a
-  -> d
-delegationCred = addrR >>> _.delegation
-
--- | Converts to cardano-serialization-lib `Address` class object.
-toAddressCsl
-  :: forall a p d
-   . Address a p d
-  => a
-  -> AddressCsl
-toAddressCsl = ToAddressCsl >>> toCslRep
-
-netTagToInt :: NetworkTag -> Int
-netTagToInt = case _ of
-  TestnetTag -> 0
-  MainnetTag -> 1
-
-intToNetTag :: Int -> Maybe NetworkTag
-intToNetTag = case _ of
-  0 -> Just TestnetTag
-  1 -> Just MainnetTag
-  _ -> Nothing
+newtype NetworkId = NetworkId Int
```

Comment:
I liked the previous representation of `NetworkTag`, what was the reason for this change? In the functions that return a `BaseAddress` below, couldn't `netTagToInt` be used?

Reply:
I decided to go for a thinner ffi that follows the library types, were kind of force to use it anyway and having only a thin wrapper means less code and types. I can change it back.

Reply:
I personally prefer the representation you had before

Reply:
Created issue that addresses this